game/debriefing.py

```diff
@@ -318,6 +322,23 @@ def dead_ground_units(self) -> GroundLosses:
                     losses.enemy_airlifts.append(airlift_unit)
                 continue
 
+        # Find killed map objects and mark them as loss if they were not added
+        # already by the killed_ground_units. This will be the case if the object id
```

Comment:
i would have changed it to the trigger_zone name right away. the problem is that this will only work for newly generated games as the TGO currently has the map_object_id only saved with it. One thing possible with a small tradeoff would be:
change completly to the trigger zone name (which gets auto generated one you click the "assign as..") for new generated games. This would mean that unit_map.buildings will only get the trigger_zone_name as the key and the TGO as Value. For 5.1.0 i could implement a save-compat workaround to use the object_id (maybe with a try and attribute error catch?) for existing campaings.
But i agree -> clean one with no IDs which could (nearly impossible but could happen) be reused by another map object

Reply:
i will look into the code snipped you mentioned above :)

Reply:
i think i was completly blinded: we do not need any object_id or name or similar anymore... The only thing we need is the trigger zone name, and the zone is already part of the TGO. We can expect the trigger_zone name to be unique. This in total would be save-compat as well. Or am i on the wrong way?

Reply:
Makes sense to me.